tcms/core/tests/test_ajax.py

```diff
@@ -33,6 +42,48 @@ def test_empty_string_is_converted_to_bool(self):
         response = self.client.get(url)
         self.assertEqual(200, response.status_code)
 
+    def test_with_ulli_format(self):
+        """ When a request comes with info_type=categories, and format=ulli for given product_id,
+            we expect to receive all categories for that product, rendered as HTML unordered list
+        """
+
+        url = '%s?info_type=categories&product_id=%d&format=ulli' % (reverse('ajax-info'),
+                                                                     self.product.pk)
+
+        response = self.client.get(url)
+
+        self.assertContains(response, '<li>' + self.category_one.name + '</li>')
+        self.assertContains(response, '<li>' + self.category_two.name + '</li>')
+
+    def test_with_json_format(self):
+        """ When a request comes with info_type=categories for given product_id,
+            we expect to receive all categories for that product as array of JSON objects """
+
+        url = "%s?info_type=categories&product_id=%d" % (reverse('ajax-info'), self.product.pk)
+
+        response = self.client.get(url)
+
+        self.assertIn(
+            json.loads('{"model": "testcases.category", "pk": %s, "fields": {"name": "%s"}}'
+                       % (self.category_one.pk, self.category_one.name)),
+            json.loads(response.content))
```

Comment:
json.loads(str(response.content, encoding=settings.DEFAULT_CHARSET)) to convert the content back to string.

Also I'm surprised assertIn doesn't raise an error here. The assert function uses `member not in container` which produces TypeError: unhashable type: 'dict' when used with dictionaries. You will have to update these assertions.